src/mame/tandberg/tdv2100_disp_logic.cpp

```diff
@@ -0,0 +1,962 @@
+}
+
+void tandberg_tdv2100_disp_logic_device::rs232_dsr_w(int state)
+{
+	bool auto_rx_handshake = m_sw_rs232_settings->read()&0x10;
+	bool dcd_handshake = ((m_dsw_u39->read()&0x00C) == 0x008);
+	if(!dcd_handshake && !auto_rx_handshake)
+	{
+		rx_handshake = state;
+	}
+	update_rs232_lamps();
+}
+
+void tandberg_tdv2100_disp_logic_device::rs232_cts_w(int state)
+{
+	bool auto_tx_handshake = m_sw_rs232_settings->read()&0x20;
+	if(!auto_tx_handshake)
+	{
+		tx_handshake = state;
+	}
+	update_rs232_lamps();
+}
+
+void tandberg_tdv2100_disp_logic_device::rs232_txd_w(int state)
+{
+	bool ext_echo = m_sw_rs232_settings->read()&0x08;
+	if(!(ext_echo && data_terminal_ready))
+	{
+		m_uart->write_si(state);
+	}
+	if(!(request_to_send && tx_handshake))
+	{
+		state = 1;
+	}
+	else if(break_key_held)
+	{
+		state = 0;
+	}
+	m_rs232->write_txd(state);
+}
+
+void tandberg_tdv2100_disp_logic_device::rs232_ri_w(int state)
+{
+	// TODO: Ring Indicator interrupt for CPU interface
+}
+
+void tandberg_tdv2100_disp_logic_device::check_rs232_rx_error(int state)
+{
+	m_write_errorl_cb(m_uart->pe_r() && m_uart->dav_r());
+}
+
+void tandberg_tdv2100_disp_logic_device::uart_rx(int state)
+{
+	if(state)
+	{
+		bool tty_mode = m_dsw_u39->read()&0x010;
+		if(tty_mode)
+		{
+			char_to_display(m_uart->receive());
+			m_uart->write_rdav(0);
+		}
+		else
+		{
+			// TODO: Rx interrupt for CPU interface
+		}
+	}
+}
+
+void tandberg_tdv2100_disp_logic_device::uart_tx(int state)
+{
+	if(state)
+	{
+		bool tty_mode = m_dsw_u39->read()&0x010;
+		if(tty_mode)
+		{
+			if(tx_data_pending_kbd)
+			{
+				m_uart->transmit(tx_data_kbd);
+				tx_data_pending_kbd = false;
+			}
+		}
+		else
+		{
+			// TODO: Tx interrupt for CPU interface
+		}
+	}
+}
+
+INPUT_CHANGED_MEMBER(tandberg_tdv2100_disp_logic_device::uart_changed)
+{
+	set_uart_state_from_switches();
+}
+
+void tandberg_tdv2100_disp_logic_device::set_uart_state_from_switches()
+{
+	int baud_setting = m_sw_rs232_baud->read();
+	int u46_divisor = (baud_setting == 0x0) ? 15 : 11;
+	int u22_divisor = 256>>baud_setting;
+	m_uart_clock->set_unscaled_clock(DOT_CLOCK_3 / (u46_divisor * u22_divisor));
+	m_uart->write_cs(true);
+	m_uart->write_nb1(m_dsw_u39->read()&0x200);
+	m_uart->write_nb2(true);
+	m_uart->write_np(m_sw_rs232_settings->read()&0x01);
+	m_uart->write_eps(m_sw_rs232_settings->read()&0x02);
+	m_uart->write_tsb(m_sw_rs232_settings->read()&0x04);
+}
+
+///////////////////////////////////////////////////////////////////////////////
+//
+// Strobes for Keyboard
+//
+
+void tandberg_tdv2100_disp_logic_device::process_keyboard_char(uint8_t key)
+{
+	if(key&0x80)
+	{
+		tx_data_kbd = key&0x7F;
+		tx_data_pending_kbd = true;
+		bool tty_mode = m_dsw_u39->read()&0x010;
+		if(tty_mode)
+		{
+			if(m_uart->tbmt_r())
+			{
+				uart_tx(1);
+			}
+		}
+		else
+		{
+			// TODO: Keyboard interrupt for CPU interface
+		}
+	}
+}
+
+void tandberg_tdv2100_disp_logic_device::w_break(int state)
+{
+	break_key_held = state;
+}
```

Comment:
Can this result in the TxD output changing immediately?

Reply:
This is the intended behaviour, the original hardware also does this.